revolv/donor/views.py

```diff
@@ -30,3 +34,78 @@ def get_context_data(self, **kwargs):
         context["donor_has_no_donated_projects"] = Project.objects.donated_projects(self.user_profile).count() == 0
 
         return context
+
+class AccountingView(UserDataMixin, TemplateView):
+    """
+    View of the accounting page.
+    """
+    template_name = 'donor/accounting.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(AccountingView, self).get_context_data(**kwargs)
+
+        context['project_names'] = [p.title for p in Project.objects.all()]
+        context['admin_repayments'] = AdminRepayment.objects.repayments()
+        context['admin_reinvestments'] = AdminReinvestment.objects.reinvestments()
+        context['donations'] = Payment.objects.donations()
+        context['jsonurl'] = unicode(str(reverse('donor:accountingPartial')))
+
+        context['clean_project_names'] = {}
+        clean_project_names = context['clean_project_names']
+        for p in context['project_names']:
+            clean_name = AccountingAggregator().clean_project_name(p)
+            clean_project_names[clean_name] = p
+
+        return context
+
+class AccountingPartialView(TemplateView):
+    """
+    A view that returns partial HTML of all relevant accounting data.
+    This is queried by an AJAX request when creating the Accounting page.
+    """
+
+    start_date = datetime.datetime.strptime('2015-01-01', '%Y-%m-%d')
+    end_date = datetime.datetime.now()
+    project_filter = Project.objects.all()
+    allProjects = Project.objects.all()
+    template_name = 'base/partials/cash_flow_table.html'
+
+    def updateFilterVariables(self, request):
+        """
+        Updates instance variables such as start_date, end_date, and project_filter
+        based on the keys passed in.
+
+        :args:
+            keys: The request passed in to the AccountingPartialView
+        """
+        if 'start_date' in request:
+            self.start_date = datetime.datetime.strptime(request['start_date'], '%Y-%m-%d')
+        elif 'end_date' in request:
+            self.end_date = datetime.datetime.strptime(request['end_date'], '%Y-%m-%d')
+        elif 'project_choice' in request:
+            if request['project_choice'] == 'all':
+                self.project_filter = Project.objects.all()
+            else:
+                project_dict = {}
+                for proj in self.allProjects:
+                    name = proj.title
+                    project_dict[AccountingAggregator().clean_project_name(name)] = name
+                real_name = project_dict[request['project_choice']]
+                self.project_filter = Project.objects.filter(title=real_name)
```

Comment:
I think it would be much nicer to have the `project_choice` be a project id instead, then you could just do a `Project.objects.get` here instead of doing all this cleaning/filtering